packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -56,7 +56,7 @@ function addVNodeToChildLWC(vnode: VCustomElement) {
 }
 
 // [h]tml node
-function h(sel: string, data: VElementData, children: VNodes): VElement {
+function h(sel: string, data: Readonly<VElementData>, children: Readonly<VNodes>): VElement {
```

Comment:
The LWC compiler generates an empty array when an element has no children. Most of the fixture outputs contain `const stcX = [];`. A preferable approach here would be to generate default the `children` argument to an empty array.

```suggestion
function h(sel: string, data: Readonly<VElementData>, children: Readonly<VNodes> = EmptyArray): VElement {
```

Same observation for the `slot` API. The `custom_element` API already does this, but it's not something the compiler leverages today.

Reply:
This is a good TODO: I'd say let's handle it in another PR: https://github.com/salesforce/lwc/issues/2671